File: Typing Speed Test/TypingSpeedTest.java
import java.util.Scanner;

public class TypingSpeedTest{
	public static int getTimeInMinutes(int userInput){
		return userInput;
	}

	public static void printResult(int userInput){
		
	}

	public static void printSentence(String input){
		String display = """

	Sentence: %s

		""";
		System.out.printf(display, input);
	}

	public static void printMainMenu(String name){
		String display = """
		
	==> %s Typing Speed Test <==

	1. Start Test
	0. Exit
		""";
	System.out.printf(display, name);
	}

	public static void main(String[] args){
		Scanner inputCollector = new Scanner(System.in);

		String sentence = "The progressive development of man";
		System.out.print("Enter your name: ");
		String userName = inputCollector.nextLine();

		String userInput = "4";
		while(!userInput.equalsIgnoreCase("0")){
			printMainMenu(userName);

			System.out.print("Make a selection: ");
			userInput = inputCollector.nextLine();
			switch (userInput){
				case "1"-> {
					printSentence(sentence);
					System.out.print("Lets Go!!!: ");
					String answer = inputCollector.nextLine();


					int matched = 0; int totalCharacter = 0;
					for (int count = 0; count < sentence.length(); count++){
						totalCharacter++;
						char sentkk = sentence.charAt(count);
						System.out.print(sentkk);
					}
					//int accuracy = (matched/totalCharacter) * 100;
					
				} // end of case 1
/*


				case "0"-> {
					System.out.print("Thank you for testing");
					break;
				}
				default-> {
					System.out.print("Please, enter a valid input");
					break;
				}
*/

			}
		}	
	}
}

File: Typing Speed Test/typing_speed_test.js
const prompt = require ("prompt-sync")({sigint:true});

let word = "Innovation"
let sentenseOne = "The progressive development of man is vitally dependent on innovation";

let userName = prompt("Enter your name: ");

	const mainDisplay = `
	
	==> ${userName} Typing Speed Test <==

	1. Start Test
	0. Exit

	`;
let userInput = 4
while(userInput != 0){
	console.log(mainDisplay);
	userInput = prompt("Enter 1 to start: ");
	switch(userInput){
		case "1": {
			const start = Date.now();
			let test = `

	Sentence : ${sentenseOne}

			`;
			console.log(test);
			let answer = prompt("Let's Go !!! : ");

			let matched = 0; let = totalCharacter = 0;
			for (let count = 0; count < sentenseOne.length; count++){
				totalCharacter++;
				if (sentenseOne[count] == answer[count]){
					matched++;
				}
			}
			let accuracy = (matched/totalCharacter) * 100;


			const end = Date.now();
			const duration = end - start;
 			
			const durationInMinutes = duration/ 60000
			const wpm = 10/durationInMinutes;
			

			console.log("Time taken: ", duration/1000,"Seconds");
			console.log("Words per minutes: ",wpm)
			console.log("Accuracy: ",accuracy)

/*

			let finalResult = `
	----------------------------------
	Result				
	----------------------------------
	Time taken: ${duration/1000}seconds
	Words per minutes: ${wpm}
	Accuracy: ${accuracy}
	----------------------------------
			`;
			console.log(finalResult);
*/
			break;
		}
		case "0": {
			console.log("Thank you for testing");
			break;
		}
		default: {
			console.log("Please, enter a valid input");
			break;
		}
	}

}


